Gems/Atom/RPI/Code/Source/RPI.Public/RPISystem.cpp

```diff
@@ -79,6 +80,17 @@ namespace AZ
 
         void RPISystem::Initialize(const RPISystemDescriptor& rpiSystemDescriptor)
         {
+            //If xr system is registered with RPI init xr instance
+            if (m_xrSystem)
+            {
+                AZ::RHI::ResultCode resultCode = m_xrSystem->InitInstance();
+                if (resultCode != AZ::RHI::ResultCode::Success)
+                {
+                    AZ_Warning("RPISystem", false, "Unable to initialize XR System");
+                }
```

Comment:
Should this be an error with an early return? 

If not, then the `if` is unnecessary:

````
[[maybe_unused]] AZ::RHI::ResultCode resultCode = m_xrSystem->InitInstance();
AZ_Warning("RPISystem", resultCode == AZ::RHI::ResultCode::Success, "Unable to initialize XR System");
````